iceoryx_posh/include/iceoryx_posh/roudi/roudi_cmd_line_parser.hpp

```diff
@@ -37,6 +37,8 @@ enum class MonitoringMode
     OFF
 };
```

Comment:
maybe the `MonitoringMode` should be moved to `iceoryx_posh_types.hpp`? It's also used in `roudi.cpp` which probably should not depend on this header

Reply:
Only `roudi_app.hpp` depends on `roudi_cmd_line_parser.hpp`. If I'd move it to `iceoryx_posh_types.hpp` we would have a free function in there. I'm happy to move it to the class `RouDiApp`. Opinions?

Reply:
Move it to `iceoryx_posh_types.hpp` and started preparing for the split of a separate `iceoryx_roudi_types.hpp` once RouDi is in its own CMake target.